examples/chooser/chooser_event_counters

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+# step 2021: The following new custom tag attributes, which
+# are ignored by older builds, gracefully degrade for backward compatibility.
+# "default-file"      overrides <default> - specify both if sensible
+# "fs-filters-mime"   like <entry>'s and listed before fs-filters
+# "fs-filters"        like <entry>'s
+
+# step 2022: A warning about GTK2 vs GTK3 signal differences
+#
+# The GTK3 GtkFileChooser triggers the "selection-changed" and "update-preview"
+# signals differently than the GTK2 GtkFileChooser.  If you want your
+# application to work on both GTK2 and GTK3 use this script to study which
+# events trigger on GTK2 vs GTK3.  Test what happens on application start, on a
+# single change of selection, and on applying file filters (fs-filters and
+# fs-filters-mime).  For the first two cases, the difference between GTK2 and
+# GTK3 is small but for file filters it is huge.  Unlike GTK2, which does not
```

Comment:
Is it possible to compensate for that difference inside gtkdialog? Many users post gtkdialog-based stuff in the forums, but they have no idea what gtkdialog version they developed against. Inconsistencies like this are exactly what must be avoided to allow new Puppy releases to migrate away from GTK+ 2.

Reply:
@dimkr I have already partially compensated [here](https://github.com/puppylinux-woof-CE/gtkdialog/blob/78f7ab33f5ed0341fc1314375127deb6a4b0bae2/src/signals.c#L2079) as part of this PR. It's a significant improvement but not a full solution.  I'm still actively trying to find a full solution but it will take time and will be a new PR, if any.